addons/html_builder/static/src/builder/snippets_menu_tabs/customize_tab.js

```diff
@@ -10,4 +11,39 @@ export class CustomizeTab extends Component {
     static defaultProps = {
         currentOptionsContainers: [],
     };
+
+    setup() {
+        this.state = useState({
+            hasContent: true,
+        });
+        const contentRef = useRef("content");
+
+        const applyVisibility = () => {
+            this.state.hasContent = [...contentRef.el.childNodes].some((el) =>
+                isTextNode(el) ? el.textContent !== "" : !el.classList.contains("d-none")
+            );
+        };
+
+        const observer = new MutationObserver(() => {
+            applyVisibility();
+        });
+        useEffect(
+            (contentEl) => {
+                if (!contentEl) {
+                    return;
+                }
+                applyVisibility();
+                observer.observe(contentEl, {
+                    subtree: true,
+                    attributes: true,
+                    childList: true,
+                    attributeFilter: ["class"],
+                });
+                return () => {
+                    observer.disconnect();
+                };
+            },
+            () => [contentRef.el]
+        );
```

Comment:
This code is mostly the same as in `useVisibleWithContent`.
Can you make an utility function?